### ISSUE TITLE
🐛 Bug Report:  int attributes in appwrite accepts numbers as String

### ISSUE THREAD

**Issue by issasaleh** [appwrite/appwrite#7810]
### 👟 Reproduction steps

1-Create a collection that have an attribute with the type int.
2-Create a function to listen to any create in the collection and log the req.body.
3- create a document in the collection with any client SDK for example  JavaScript :

```
databases.createDocument(
        'DatabaseId,
        'CollectionId',
        {
            "AMOUNT": "50000"
        }
    );
```

4- pass the int variable as String like this : "AMOUNT": "50000".
5- go look in the function logs to get the req.body you will find it as String although the attribute is int.

### 👍 Expected behavior

the create document should throw an error because the type is not int.

### 👎 Actual Behavior

everything went fine, the document got created successfully. and the function logs print the attribute as a **_String_** !!
this will cause problems in the function if I'm expecting to get an int value from the function event but got **_String_** instead.

![111](https://github.com/appwrite/appwrite/assets/114824732/57592e6d-b558-4f61-af1a-22e4129be3af)

![response](https://github.com/appwrite/appwrite/assets/114824732/9a37c637-7a5a-49a6-985d-9fd0661bfffb)

Discord thread: https://discord.com/channels/564160730845151244/1217123986018664488

### 🎲 Appwrite version

Appwrite Cloud

### 💻 Operating system

Windows

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by FatumaA** [comment]
It seems you have passed the amount a string value so it inferred it as a string, did you explicitly set the amount as int in appwrite? or set the type of amount to be an int type somewhere else in the code?

**Comment by IOAPP-IO** [comment]
> It seems you have passed the amount a string value so it inferred it as a string, did you explicitly set the amount as int in appwrite? or set the type of amount to be an int type somewhere else in the code?

The AMOUNT attribute is an int in appwrite, it should never accepts Numbers as strings, ever.

But from what i see it accepts numbers as Strings then maybe it got parsed in the process in appwrite before it reaches the database **but**__ the function event catch it before it got parsed, and this is the issue with that, if a programming error happens in the front end and the programer pass the value as string, appwrite should throw an error, 

If im expecting to get an int value from the function event but got String insted that will cause problems.

Why parse the value then write it to the database when you can make it only accepts int simply.

**Comment by IOAPP-IO** [comment]
> It seems you have passed the amount a string value so it inferred it as a string, did you explicitly set the amount as int in appwrite? or set the type of amount to be an int type somewhere else in the code?

Try the reproducing steps that i explained above and you will see that.

**Comment by issasaleh** [comment]
> It seems you have passed the amount a string value so it inferred it as a string, did you explicitly set the amount as int in appwrite? or set the type of amount to be an int type somewhere else in the code?


The AMOUNT attribute is an int in appwrite, it should never accepts Numbers as strings, ever.

But from what i see it accepts numbers as Strings then maybe it got parsed in the process in appwrite before it reaches the database *but*__ the function event catch it before it got parsed, and this is the issue with that, if a programming error happens in the front end and the programer pass the value as string, appwrite should throw an error, 

If im expecting to get an int value from the function event but got String insted that will cause problems.

Why parse the value then write it to the database when you can make it only accepts int simply.

**Comment by issasaleh** [comment]
> It seems you have passed the amount a string value so it inferred it as a string, did you explicitly set the amount as int in appwrite? or set the type of amount to be an int type somewhere else in the code?

try the Reproduction steps...

**Comment by FatumaA** [comment]
Ok, I have reproduced this bug. It's a bit nuanced. There is no error from Appwrite because it turns your string into an int but returns a string in the create event response. If you then fetch the documents it will return the new entry with the correct type. 

My thoughts are that Appwrite should either:
- Throw an error when you pass anything that is not an int
- Or it should return the create response with the attribute as an int after it parses it to int type in the background

**Comment by issasaleh** [comment]
> Ok, I have reproduced this bug. It's a bit nuanced. There is no error from Appwrite because it turns your string into an int but returns a string in the create event response. If you then fetch the documents it will return the new entry with the correct type.
> 
> My thoughts are that Appwrite should either:
> 
> * Throw an error when you pass anything that is not an int
> * Or it should return the create response with the attribute as an int after it parses it to int type in the background

Yes,

Because the "Create event" returns the data before parsing it, it will cause issues if I expect an int from `request.body` in the function and get a string instead.

I think the most effective solution is the first one you suggested:

- Throw an error when you pass anything that is not an int

**Comment by stnguyen90** [comment]
> Throw an error when you pass anything that is not an int

Yes, I agree we should throw an error if the incorrect type is passed. 

Would either of you be interested in working on this?

**Comment by FatumaA** [comment]
I'd love to give it a shot

**Comment by abnegate** [comment]
@stnguyen90 This is intentional, it would just add unnecessary friction to throw an error if the string is a valid integer. The event should definitely fire with with an int though.

**Comment by IOAPP-IO** [comment]
> @stnguyen90 This is intentional, it would just add unnecessary friction to throw an error if the string is a valid integer. The event should definitely fire with with an int though.

The it should return the create response with the attribute as an int after it parses it to int type in the background..

**Comment by stnguyen90** [comment]
@abnegate, so where and how would be the best place to convert the attribute such that we don't miss anything and we handle all attribute? The [Structure validation](https://github.com/utopia-php/database/blob/537c555674ce3edeccecf323ba9361f1cb84ee96/src/Database/Database.php#L3064) should only be for validation and shouldn't mutate the data, right? Would we need to update [encode()](https://github.com/utopia-php/database/blob/537c555674ce3edeccecf323ba9361f1cb84ee96/src/Database/Database.php#L3060)?

**Comment by HarshMN2345** [comment]
This works on main now. Closing, but reopen it if you still hit it.
